File: app/src/main/java/com/devdream/cookall/main/MainPresenter.java
package com.devdream.cookall.main;

import android.view.View;
import android.widget.Toast;

import com.devdream.cookall.core.interactors.RecipeInteractor;

public class MainPresenter implements MainListener {

    private MainActivity mainActivity;

    private RecipeInteractor recipeInteractor;

    public MainPresenter(MainActivity _mainActivity) {
        mainActivity = _mainActivity;
        recipeInteractor = new RecipeInteractor();
    }

    @Override
    public void shareRecipe(View view) {
        Toast.makeText(mainActivity, "Shared!", Toast.LENGTH_SHORT).show();
    }

    @Override
    public void likeRecipe(View view) {
        Toast.makeText(mainActivity, "Liked!", Toast.LENGTH_SHORT).show();
    }

    @Override
    public void favoriteRecipe(View view) {
        Toast.makeText(mainActivity, "Favorited!", Toast.LENGTH_SHORT).show();
    }

}

File: app/src/main/java/com/devdream/cookall/main/MainPresenter.kt
package com.devdream.cookall.main

import android.view.View
import android.widget.Toast

import com.devdream.cookall.core.interactors.RecipeInteractor

class MainPresenter(private val mainActivity: MainActivity) : MainListener {

    private val recipeInteractor: RecipeInteractor

    init {
        recipeInteractor = RecipeInteractor()
    }

    override fun shareRecipe(view: View) {
        Toast.makeText(mainActivity, "Shared!", Toast.LENGTH_SHORT).show()
    }

    override fun likeRecipe(view: View) {
        Toast.makeText(mainActivity, "Liked!", Toast.LENGTH_SHORT).show()
    }

    override fun favoriteRecipe(view: View) {
        Toast.makeText(mainActivity, "Favorited!", Toast.LENGTH_SHORT).show()
    }

}
